### PR TITLE
Infinitescroll

### PR DESCRIPTION
**Pull request by amrathmd**

### CODE REVIEW

File: src/controllers/job.controller.js [hunk 1]
```diff
@@ -7,17 +7,17 @@ const CreateJob= async (req,res)=>{
...
-  const Id = req.cookies.user;
+  const page= req.query.page? parseInt(req.query.page) : 0;
+  const jsonObject=await jobService.getJob(page);
   res.send(jsonObject);
 }
-const getJobCreatedById=async (req,res)=>{
-  const userId=req.params.user;
-  const jobs= await jobService.getJobById(userId);
+const getJobCreatedByUser=async (req,res)=>{
+  const userId=req.query.email;
```
Comment (vadlachary2002) [inline src/controllers/job.controller.js]: we can use cookies for userId

File: src/models/job.model.js [hunk 2]
```diff
@@ -50,10 +50,26 @@ const jobSchema = new Schema({
   createdBy:{
     type: String,
     required: true,
+    unique: true,
```
Comment (vadlachary2002) [inline src/models/job.model.js]: remove unique a user can post multiple jobs created by must not unique
make a pull from develop

File: src/models/job.model.js [hunk 3]
```diff
@@ -50,10 +50,26 @@ const jobSchema = new Schema({
...
+    validate(value) {
+      if (!validator.isEmail(value)) {
+        throw new Error('Invalid email');
+      }
+    },
   },
   updatedBy:{
     type: String,
     required: true,
+    unique: true,
```
Comment (vadlachary2002) [inline src/models/job.model.js]: remove unique

File: src/models/userModel.js [hunk 4]
```diff
@@ -25,8 +25,8 @@ const UserSchema = mongoose.Schema({
   },
   role :{
     type:String,
-    enum:['read-only','read-write','admin'],
-    default:"read-only",
+    enum:['USER','ADMIN','SUPER_ADMIN'],
```
Comment (vadlachary2002) [inline src/models/userModel.js]: the update was merged please pull from develop

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,5 @@
 node_modules
 .idea
 .env
+.DS_Store
```
Comment (vadlachary2002) [inline .gitignore:4]: .DS_Store file  can be removed

**Review by maharjunm — APPROVED** [review]

**Pull request merged**